orderer/common/cluster/replication.go

```diff
@@ -663,11 +663,17 @@ func ExtractGenesisBlock(logger *flogging.FabricLogger, block *common.Block) (st
 	metadata := &common.BlockMetadata{
 		Metadata: make([][]byte, 4),
 	}
+	metadata.Metadata[common.BlockMetadataIndex_SIGNATURES] = protoutil.MarshalOrPanic(&common.OrdererBlockMetadata{
+		LastConfig: &common.LastConfig{Index: 0},
+		// This is a genesis block, peer never verify this signature because we can't bootstrap
+		// trust from an earlier block, hence there are no signatures here.
+	})
 	metadata.Metadata[common.BlockMetadataIndex_LAST_CONFIG] = protoutil.MarshalOrPanic(&common.Metadata{
 		Value: protoutil.MarshalOrPanic(&common.LastConfig{Index: 0}),
```

Comment:
Because of the removed check on line 514, this could now result in an out of bounds dereference.  We either need to restore the original check, or, add an additional check before this dereference.